rs/cli/src/cli.rs

```diff
@@ -484,5 +487,198 @@ pub mod proposals {
             #[clap(long)]
             omit_large_fields: Option<bool>,
         },
+
+        /// More user-friendly command for filtering proposals
+        Filter {
+            /// Limit on the number of \[ProposalInfo\] to return. If value greater than
+            /// canister limit is used it will still be fetch the wanted number of proposals.
+            #[clap(long, default_value = "100")]
+            limit: u32,
+
+            /// Proposal statuses to include. If not specified will include all proposals.
+            #[arg(value_enum)]
+            #[clap(long, aliases = ["status"], short = 's')]
+            statuses: Vec<ProposalStatus>,
+
+            /// Proposal topics to include. If not specified will include all proposals.
+            #[arg(value_enum)]
+            #[clap(long, aliases = ["topic"], short = 't')]
+            topics: Vec<Topic>,
+        },
+    }
+
+    #[derive(ValueEnum, Clone, Debug)]
+    pub enum ProposalStatus {
+        Unspecified = 0,
+        /// A decision (adopt/reject) has yet to be made.
+        Open = 1,
+        /// The proposal has been rejected.
+        Rejected = 2,
+        /// The proposal has been adopted (sometimes also called
+        /// "accepted"). At this time, either execution as not yet started,
+        /// or it has but the outcome is not yet known.
+        Adopted = 3,
+        /// The proposal was adopted and successfully executed.
+        Executed = 4,
+        /// The proposal was adopted, but execution failed.
+        Failed = 5,
+    }
+
+    impl From<ProposalStatus> for ProposalStatusUpstream {
+        fn from(value: ProposalStatus) -> Self {
+            match value {
+                ProposalStatus::Unspecified => Self::Unspecified,
+                ProposalStatus::Open => Self::Open,
+                ProposalStatus::Rejected => Self::Rejected,
+                ProposalStatus::Adopted => Self::Adopted,
+                ProposalStatus::Executed => Self::Executed,
+                ProposalStatus::Failed => Self::Failed,
+            }
+        }
+    }
+
+    impl From<ProposalStatusUpstream> for ProposalStatus {
+        fn from(value: ProposalStatusUpstream) -> Self {
+            match value {
+                ProposalStatusUpstream::Unspecified => Self::Unspecified,
+                ProposalStatusUpstream::Open => Self::Open,
+                ProposalStatusUpstream::Rejected => Self::Rejected,
+                ProposalStatusUpstream::Adopted => Self::Adopted,
+                ProposalStatusUpstream::Executed => Self::Executed,
+                ProposalStatusUpstream::Failed => Self::Failed,
+            }
+        }
+    }
+
+    #[derive(ValueEnum, Clone, Debug)]
+    pub enum Topic {
```

Comment:
I suppose it wasn't possible to use the upstream Topic? What was the show stopper?

Reply:
`ValueEnum` trait is not implemented for the upstream Topic, and i cannot manually implement a trait from a different crate (clap) for an enum/struct from a third crate (ic)